src/test/ui/generic-associated-types/method-unsatified-assoc-type-predicate.stderr

```diff
@@ -1,23 +1,18 @@
-error[E0599]: the method `f` exists for struct `S`, but its trait bounds were not satisfied
+error[E0599]: no method named `f` found for struct `S` in the current scope
   --> $DIR/method-unsatified-assoc-type-predicate.rs:30:7
    |
 LL | struct S;
-   | --------
-   | |
-   | method `f` not found for this struct
-   | doesn't satisfy `<S as X>::Y<i32> = i32`
-   | doesn't satisfy `S: M`
+   | -------- method `f` not found for this struct
 ...
 LL |     a.f();
-   |       ^ method cannot be called on `S` due to unsatisfied trait bounds
+   |       ^ method not found in `S`
    |
-note: trait bound `<S as X>::Y<i32> = i32` was not satisfied
-  --> $DIR/method-unsatified-assoc-type-predicate.rs:14:11
+   = help: items from traits can only be used if the trait is implemented and in scope
+note: `M` defines an item `f`, perhaps you need to implement it
+  --> $DIR/method-unsatified-assoc-type-predicate.rs:10:1
    |
-LL | impl<T: X<Y<i32> = i32>> M for T {}
-   |           ^^^^^^^^^^^^   -     -
-   |           |
-   |           unsatisfied trait bound introduced here
```

Comment:
This would qualify as a regression, right?

Reply:
IMO this suggests that it's the right error:
```
note: `M` defines an item `f`, perhaps you need to implement it
```

At most we could show the `impl` as "hey btw this `M` impl exists but doesn't apply here" (which I think we do sometimes? maybe not for blanket impls? or is it for an error that's unambiguous about which trait is, and just not method call resolution failures?)

Reply:
But in this case it *is* a blanket impl, which means that the user *can't* impl M themselves due to orphan rules, right?

Reply:
As an analogy (to avoid the `M`/`X` names here), you can implement `IntoIterator` yourself on your own types, the only time it's disallowed is if you *also* implemented `Iterator` (in which case you have `IntoIterator` already handled by the blanket `impl` so the error isn't a concern).

If the type wasn't a locally-defined `struct S;`, maybe I could see "perhaps you need to implement it" being suboptimal, but it's fine.